src/Core/Domain/Discount/Command/UpdateDiscountCommand.php

```diff
@@ -0,0 +1,366 @@
+            $this->localizedNames = null;
+        }
+        foreach ($localizedNames as $languageId => $name) {
+            $this->localizedNames[(new LanguageId($languageId))->getValue()] = $name;
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return array<int, string>|null
+     */
+    public function getLocalizedNames(): ?array
+    {
+        return $this->localizedNames;
+    }
+
+    public function isActive(): ?bool
+    {
+        return $this->active;
+    }
+
+    public function setActive(?bool $active): self
+    {
+        $this->active = $active;
+
+        return $this;
+    }
+
+    public function getValidFrom(): ?DateTimeImmutable
+    {
+        return $this->validFrom;
+    }
+
+    public function getValidTo(): ?DateTimeImmutable
+    {
+        return $this->validTo;
+    }
+
+    /**
+     * @throws DiscountConstraintException
+     */
+    public function setValidityDateRange(?DateTimeImmutable $from, ?DateTimeImmutable $to): self
+    {
+        if (null === $from || null === $to) {
+            throw new DiscountConstraintException('Cannot validate range if one of properties is null', DiscountConstraintException::INVALID_DATE_FROM_OR_DATE_TO);
+        }
+        $this->assertDateRangeIsValid($from, $to);
+        $this->validFrom = $from;
+        $this->validTo = $to;
+
+        return $this;
+    }
+
+    public function getPriority(): ?int
+    {
+        return $this->priority;
+    }
+
+    /**
+     * @throws DiscountConstraintException
+     */
+    public function setPriority(?int $priority): self
+    {
+        if (null === $priority || 0 >= $priority) {
+            throw new DiscountConstraintException(
+                sprintf('Invalid discount priority "%s". Must be a positive integer.', $priority),
+                DiscountConstraintException::INVALID_PRIORITY
+            );
+        }
+
+        $this->priority = $priority;
+
+        return $this;
+    }
+
+    public function getTotalQuantity(): ?int
+    {
+        return $this->totalQuantity;
+    }
+
+    public function isHighlightInCart(): ?bool
+    {
+        return $this->highlightInCart;
+    }
+
+    public function setHighlightInCart(?bool $highlightInCart): void
+    {
+        $this->highlightInCart = $highlightInCart;
+    }
+
+    /**
+     * @throws DiscountConstraintException
+     */
+    public function setTotalQuantity(?int $quantity): self
+    {
+        if (null === $quantity || 0 > $quantity) {
+            throw new DiscountConstraintException(sprintf('Quantity cannot be lower than zero, %d given', $quantity), DiscountConstraintException::INVALID_QUANTITY);
+        }
+
+        $this->totalQuantity = $quantity;
+
+        return $this;
+    }
+
+    public function getQuantityPerUser(): ?int
+    {
+        return $this->quantityPerUser;
+    }
+
+    /**
+     * @throws DiscountConstraintException
+     */
+    public function setQuantityPerUser(?int $quantity): self
```

Comment:
```suggestion
    public function setQuantityPerUser(?int $quantityPerUser): self
```
If you can make the parameters match the class fields it's more in tune with our usual convention